crates/ast_node/src/encoding/mod.rs

```diff
@@ -0,0 +1,45 @@
+pub(crate) mod decode;
+pub(crate) mod encode;
+
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+enum EnumType {
+    Unit,
+    One,
+    Struct,
+}
+
+pub(crate) fn is_unknown(attrs: &[syn::Attribute]) -> bool {
+    attrs
+        .iter()
+        .filter(|attr| attr.path().is_ident("encoding"))
```

Comment:
Can this function detect attributes like

```rust
    #[cfg_attr(
        feature = "encoding-impl",
        encoding(with = "cbor4ii::core::types::Maybe")
    )]
```
?

Reply:
I'm not sure if it's required. Is detection via this function (in the proc-macro) important for the operation?

Reply:
Do you want to automatically check the `Option` in a macro? I think that's hard because macros are not type-aware.

I think explicit attributes are simpler and more reliable.